docs/whatsnew/6.1.rst

```diff
@@ -34,6 +34,32 @@ on Windows and potentially 32-bit architectures. Previously on those platforms,
 columns with any long integers which overflowed the 32-bit integer would be returned
 as string columns. The new default behavior is consistent with ``numpy`` v2 and ``pandas``.
 
+.. _whatsnew-6.1-quantity-format-spec:
+
+``Quantity.to_string`` supports ``formatter`` for formatting
+==============================================================
+
+The :meth:`~astropy.units.Quantity.to_string` method now supports a ``formatter`` parameter.
+This feature supports both Python's format specification mini-language via format strings and
+custom formatting through callables. This enables users to have explicit and consistent control
+over the numerical representation of quantities, accommodating a wide range of formatting needs.
+
+Previously, the method primarily relied on the ``precision`` parameter for format control, which dictated
+the number of significant digits and did not provide much freedom in the latex format.
+
+Example:
+
+.. code-block:: python
+
+    from astropy import units as u
+    q = u.Quantity(123.456, u.m)
+    custom_format = lambda x: f"\\approx {float(x):.1f}",
+    print(q.to_string(formatter=custom_format, format='latex'))
+    # Outputs: "$\approx 123.5 \; \mathrm{m}$",
+    print(q.to_string(formatter='.3e', format='latex'))
+    # Outputs: '$1.235 \times 10^{2} \; \mathrm{m}$'
+    print(q.to_string(precision=3, format='latex'))
+    # Outputs: '$123 \; \mathrm{m}$'
```

Comment:
This code example needs to be rewritten in its entirety, both to match other code examples in style and to ensure the example is actually tested.

Reply:
Hmm, you're right - when I saw this I had this vague memory that we do not run doctests on what's-new, but clearly we do write them as such - @Deen-dot, see https://docs.astropy.org/en/v6.0.0/whatsnew/6.0.html for examples (no `.rst` any more since we don't guarantee those stay the same, but it used to be an `.rst` file too).

Reply:
It appears the What's New entries are indeed not tested. Including the entries for the next release in the tests would be a good idea, but doing that goes beyond the scope here. However, writing the code example in the same style as all the others is a good idea regardless.

Reply:
doctestplus only picks up things with `>>>`.